runtime/compiler/optimizer/J9RecognizedCallTransformer.cpp

```diff
@@ -1694,13 +1694,25 @@ bool J9::RecognizedCallTransformer::isInlineable(TR::TreeTop* treetop)
          case TR::java_lang_Long_rotateRight:
             return comp()->target().cpu.getSupportsHardware64bitRotate();
          case TR::java_lang_Integer_compress:
-            return comp()->target().cpu.getSupportsHardware32bitCompress();
+            {
```

Comment:
@Spencer-Comin  When I recommended the environment option to disable recognition of, I had an idea of disabling all set of compress and expand intrinsic - That being said, I am ok with this as well as long as it is possible to do that, it is just that to completely disable only expand / compress we would need to set 4 environment variable. 

We should have a way for code-gen to set support for this or not (Instead of RecognizedCallTransformer to query CPU directly.

Reply:
I think the direct CPU query (similar to the rotate intrinsic queries above) is appropriate here. The codegen check for support would just be a wrapper around the CPU query.

I could reduce the environment variables here to one that disables all four intrinsics.

Reply:
Well my concern mainly here was that this operation does not directly map to single instruction (Compress/Expand) on Z. Evaluators actually performs additional operations to match the implementation of Long/Integer.compress/expand instrinsic. 

But yes, if we can reduce number of environment variables (One for Compress and One for Expand) that would work as well.

Reply:
Oh that's a good point. In that case I'll rework this to have codegen flags for support.